### PR TITLE
[MC-102] Add cloud storage interface

### PR DESCRIPTION
**Pull request by eliasscaceres**
## What did you implement:

Closes #XXXXX

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [ ] Write tests and confirm existing functionality is not broken.
       **Validate via `npm test`**
- [ ] Write documentation
- [ ] Ensure there are no lint errors.
       **Validate via `npm run lint-updated`**
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [ ] Ensure introduced changes match Prettier formatting.
       **Validate via `npm run prettier-check-updated`**
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

**_Is this ready for review?:_** NO
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: gcp/src/services/gcpStorage.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  WriteBlobOptions,
+  convertToStream,
+  Guard,
+  WriteBlobOutput,
+} from "@multicloud/sls-core";
+import { Stream } from "stream";
+import { injectable } from "inversify";
+
+/**
+ * Implementation of CloudStorage for AWS S3 Storage
```
Comment (taten) [inline gcp/src/services/gcpStorage.ts]: ;)

File: gcp/src/services/gcpStorage.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ */
+@injectable()
+export class GcpStorage implements CloudStorage {
+  private storage: Storage;
+
+  /**
+   * Initialize new AWS S3 service
```
Comment (taten) [inline gcp/src/services/gcpStorage.ts]: AWS is taking over the world

**Review by taten — APPROVED** [review]
Looks good! Left a couple things about the code comments.

File: gcp/src/services/gcpStorage.test.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
+import { GcpStorage } from ".";
+import { Storage } from "@google-cloud/storage";
+import { Readable, Writable } from "stream";
```
Comment (FabioOrlinski) [inline gcp/src/services/gcpStorage.test.ts]: You are not using the "Writable" function. Could it be deleted?

File: gcp/src/services/gcpStorage.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      readStream
+        .pipe(
+          file.createWriteStream({
+            resumable: false,
+            validation: false,
+            metadata: { "Cache-Control": "public, max-age=31536000" },
+          })
+        )
+        .on("error", (error) => {
+          console.log("ERROR");
```
Comment (FabioOrlinski) [inline gcp/src/services/gcpStorage.ts]: Be careful with the console.log! :)

**Review by FabioOrlinski — APPROVED** [review]

**Pull request merged**